# Problem Set 1 (100 points + 15 bonus points)

### Due: October 7th 2020, 6:30pm

---

This problem set will test your understanding on Python Basics (Sessions 2 - 3) and the `pandas` and `numpy` packages (Session 4). It also includes a couple of questions on your final project. 

Please carefully review our Jupyter Notebooks and [class exercises](https://www.notion.so/Data-790-Solutions-to-Class-Exercises-88b6957852274c288fd004f88400c5c2) before you start working on the problems. You may also come to my [office hours](https://app.acuityscheduling.com/schedule.php?owner=20428964) if you are still struggling with some parts of the course content. 

For each of the problems, the rubric for grading is as following:

|   Grade     |   Description    |
|:-------|:------|
| Full points| The program is bug free and does everything the prompt asks for, and it includes appropriate comments and docstring (for functions).|
| 80% - 99% of the points | The program does most things the prompt asks for and is more or less bug free, and it includes  comments/docstring.       |
| 50% - 79% of the points | The program has some *hidden / occasional* bugs. It may fail some, but not all, test cases.  |
| less than 50% of the points | The program is buggy and cannot run (throws an error), but you showed some amount of understanding of the concepts.|
| 0 point | Did not provide any solution or showed little to zero understanding of the concepts. |

**Note 1: Please refer to *Lecture 3 Jupyter Notebook* for the definitions of *hidden/occasional* bugs, and to learn about how to debug your programs. You may want to test the functionalities of your programs with a few test cases before you submit your answers.** 

**Note 2: Each problem may have multiple solutions.**

**Note 3: Since this is our first pset, I left hints for most of the problems to help Python beginners. However, if you are somewhat familiar with Python, please try to solve the problems on your own unless you really need some clues.**

<div class = "alert alert-warning">
 

You may work collaboratively with other classmates (and you are encouraged to use the Discussion Board). However, you should **clearly indicate whom you work/consult with** for the specific problems that you worked on with others. (Just leave a comment or Markdown note for that specific problem(s).) <br>

You should **never** copy and paste others' code. Problem sets without proper acknowledgement violates [class policies](https://www.notion.so/DATA-790-Data-Visualization-with-Python-bebb5931ca584d0f904e7d6c64010d7d), and plagiarism will lead to a zero mark. <br>
    
</div>

---
## Q1. Control Flows and Simple functions (10 points)

Let's first warm up by writing a couple of simple programs involving control flows (the problems are adapted from our class exercises.) <br>

Control flow and funtion are the most basic set-up for many complicated programs, and you will use them a lot throughout this pset.

### A. Check vowels and consonants (5 pts)

Please write a small peice of control flow to check and print out the number of vowels and the number of consonants in a string `s` = `Welcome to DATA seven ninety Data Visualization with Python`.

**[Double click here to see hints]**

<!----

1. Use a for loop and if statements.
2. Python is case sensitive, such that `a` != `A`.

--->

In [10]:
s = 'Welcome to DATA seven ninety Data Visualization with Python'
vows = ['a', 'e', 'i', 'o', 'u']

## your code below:

for s in 'Welcome to DATA seven ninety Data Visualization with Python':
     if s in vows:
        print("vowel")
    
     else:
        print("consonant")







consonant
vowel
consonant
consonant
vowel
consonant
vowel
consonant
consonant
vowel
consonant
consonant
consonant
consonant
consonant
consonant
consonant
vowel
consonant
vowel
consonant
consonant
consonant
vowel
consonant
vowel
consonant
consonant
consonant
consonant
vowel
consonant
vowel
consonant
consonant
vowel
consonant
vowel
vowel
consonant
vowel
consonant
vowel
consonant
vowel
vowel
consonant
consonant
consonant
vowel
consonant
consonant
consonant
consonant
consonant
consonant
consonant
vowel
consonant


### B. Odd or Even (5 pts)
Please write a small function `odd_even()`, which takes in a number `n` (int or float) and returns the string, `"The number is odd."`, if the number is odd, returns the string, `"The number is even."`, if the number is even, and returns the string, `"The number is a float."`, if the number is a float.  <br>

**[Double click here to see hints]**

<!---

1. Use the remainder of n divided by 2 to check whether a number is odd or even. 
2. use `%` to calculate remainder.

--->

In [6]:
def odd_even(n):
    '''
    add docstring here
    '''
    ## your code here
    
    if (n % 2) == 0:
        print("The number is even.")
    else:
        print("The number is odd.")
    
    return n
odd_even(4)
    
    

The number is even.


4

## Q2. Operations on lists (5 points)

A list is one of the most commonly used "container" types in Python. 

Please write a function `list_oper()`, which take as input two lists `a_list` ($[a_1, a_2, ..., a_n]$) and `b_list` ($[b_1, b_2, ..., b_n]$), both of the same length, and computes

$$2a_i + b_i^2$$

for each $a_i$,  $b_i$ in the two lists, then returns a list of the returned values ($[2a_1+b_1^2, 2a_2+b_2^2, ..., 2a_n+b_n^2]$).

You may assume both lists are of the same length (so your code does not need to check for this). 

**[Double click here to see hints]**

<!---

1. You could use a `zip()` function. 
2. Alternatively, you could use `range()` and `len()` to loop over sequences. 

--->

In [15]:
def list_oper(a_list, b_list):
    '''
    add docstring here
    '''
    ## your code here
a_list = [10, 30, 50, 70, 90]
b_list = [20, 40, 60, 80]
list(zip(a_list, b_list))
    
    
    
    
    
    


[(10, 20), (30, 40), (50, 60), (70, 80)]

## Q3. Convert binary to decimal, and vice versa (10 points + 10 bonus points)

When we encountered the floating-point errors during class, we talked about the binary representation of numbers in computer memory. You know that all numbers are stored as base-2 numbers in computers. 

### A. Let's first consider the conversion from binary numbers to integers. (10 pts)

Please write a function `b2d()`, which takes a binary number `b` (base 2) as input, converts the number to its decimal form (base 10), and returns the decimal number.  

#### How to convert binary to base 10?

[This webpage](https://www.rapidtables.com/convert/number/binary-to-decimal.html) describes the steps to convert binary to int. Please read the steps and write a Python program accordingly. 

For example, if the input is a binary `1011`, your function should return the integer `11` ($1\times2^3+0\times2^2+1\times2^1+1\times2^0=11$). If the input is a binary `10011`, your function should return the integer `19` ($1\times2^4+0\times2^3+0\times2^2+1\times2^1+1\times2^0=19$). 

**You only need to consider positive integers when writing the program.**

*Note: please write your own function and do not use the built-in function `int(b, base=2)` to convert binary to int.*

**[Dpuble click here to see hints]**

<!--------
1. You could convert the binary number `b` to a str in order to deal with each digit one by one. 
2. Then how to do the necessary calculations if the digits are `str` type? Just convert them back. 
3. Alternatively, you could use arithmatic operations to access each digit one by one. 
------>

In [87]:
def b2d(b):
    '''
    add docstring here
    '''
    ## your code here
    
    d = 0
    b = list(str(b))
    b = b[::-1]
    p = 0 
    for n in b:
        if n == '1':
            d += p
        p += 1
    return d

b2d(10011)  
    

5

### B. Now let's try to convert decimal to binary. (Optional, 10 bonus pts)

**This problem is optional, but it is not difficult. It just requires the use of a `while` loop. You may read Lecture 3 Jupyter Notebook to learn about `while` loops, and earn up to 10 bonus points if you solve this problem.**

Please write a function `d2b()` which takes **a positive integer `n`** (base 10) as input, converts the number to its binary form (base 2), and returns the binary number as a **string**. 

*Note: please write your own function and do not use the built-in function `bin()` or `format()` to convert int to binary.*

#### How to convert a decimal number (integer) to a binary number?

[This webpage](https://www.rapidtables.com/convert/number/how-decimal-to-binary.html) describes the convertion process in four steps. Please refer to the steps to write your function. 

For example, if the input is an integer `13`, your function should return a string `1101`.

If the input is an integer `174`, your function should return a string `10101110`. 

**[Double click here to see hints]**


<!-----

1. Use `//` to calculate the integer quotient; Use `%` to calculate the remainder. 
2. when you return the final outcome, you could use the string method `.join()` to turn a list of numbers into a string of digits. But remember that .join() only works on `str` type. (see Lecture 3 notebook.)
3. for advanced Python users: a `while` loop is useful here. But possible solutions are not limited to it. You could also use a recursive function. 

--->

In [63]:
def d2b(n):
    '''
    add docstring here
    '''
    ## your code here
    
    if(n > 1):
        d2b(n//2)
    else:
        return(n)
d2b(13)
    
    
    
    

## Q4. Pig Latin (10 points)

In this question, we will work on converting words into a simpler version of [Pig Latin](https://en.wikipedia.org/wiki/Pig_Latin). 

In particular, please write a function `pig_latin()` which takes a string `s` as input, and converts the string to a Pig Latin  following two rules below:

1. if a string begins with a consonant, all consonants before **the first vowel** are moved to the end of the word and the letters `yay` are then added to the end, e.g. "data" becomes "atadyay" or "smile" becomes "ilesmyay".
2. if the string begins with a vowel, then "hay" is added to the end, e.g. "apple" becomes "applehay". 

Your function `pig_latin()` will then return the Pig Latin version of the input string. You may assume the input is a **lowercase string**. 

**[Double click here to see hints]**

<!---
Note that you should only move the consonants before the first vowel. This means your program should be able to `break` when it hits the first vowel in the string. 
---->

In [38]:
sentence = "apple"

In [39]:
sentence

'apple'

In [42]:
for a in sentence:
    print(a)

a
p
p
l
e


In [43]:
a

'e'

In [81]:
def pig_latin(s):    
    '''
 add docstring here
    '''
    vow = ['a', 'e', 'i', 'o', 'u']
    
    ## your code here
    for a in s:
        if a[0] in vow:
            print(a+ "ay")
        else:
            print(a[1:]+a[0]+ "ay")
    return s

pig_latin("apple")



aay
pay
pay
lay
eay


'apple'

## Q5. Longest substring (10 points)

Let's continue to play around with `str`.

Please write a function `long_sub()` which takes a *lowercase* string `s` as input, and returns the longest substring of `s` that is in alphabetical order. 

For example, if `s = fdhakgbaaekvadsnfda`, then your function should return:
```python
'aaekv'
```

If there are ties, your function should return the first substring. For example, if `s = bcdabcaoi`, then your function should return:

```python
'bcd'
```    

*Note:* This question may be difficult. If you spend more than one hour on it, I would suggest that you move on to other parts of the pset and come back later. When some programs are challenging, we should take breaks and do not try to crack it in one sitting. 

I left a few important hints in the cell below. Feel free to check them out. 

**[Double click here to see hints]**

<!---

1. Use `<` `>` on strings to check alphabetical order. 

2. According to Python slice notation, s[1] is equivalent to s[1:2].

3. You may encounter an IndexError if your index is out of range. However, as we emphasized during class, some expressions, e.g. s[-1:0] will return an empty string '' instead of raising an error. 

---->

In [102]:
def long_sub(s):
    '''
    add docstring here
    '''
    ## your code here
    ## Rachel was helping me with this question.
    
    prevChar = ""
    currentString = ""
    longestString = ""
    
    for s in string:
        if prevChar <= s:
            currentString += s
            if len(currentString) > len(longestString):
                    longestString = currentString
        else:
            currentString = s
        prevChar = s
    
print(longestString)
    
    
    
    
    
    

NameError: name 'longestString' is not defined

## Q6. NumPy Exercises (10 points)

Now let's turn to practicing the use of two important packages, `numpy` and `pandas`. 

First, you will solve a series of short NumPy exercises. 

### A. Redo Q2, but this time use NumPy. `a_list` and `b_list` can be either `np.array` or `list`. Make sure your return is a `np.array`.  (3 pts)

In [93]:
## first, import the `numpy` package: 

import numpy as np
## then, define your function:

def np_list_oper(a_list, b_list):
    '''
    add docstring
    '''
    ## your code here
  
    a_list = np.array([10, 30, 50, 70, 90])
    b_list = np.array([20, 40, 60, 80])
 
    
    return(np_list_oper)
    
np_list_oper(a_list, b_list)

NameError: name 'a_list' is not defined

### B. Write a function `all_even` that takes in a `np.array`, `arr` and returns a `np.array` of all values in `arr` that are even numbers.  (3 pts)

In [16]:
def all_even(arr):
    '''
    add docstring
    '''
    ## your code here
    import numpy as np  

    arr = np.array([n for n in arr if n % 2 == 0])

    return arr

all_even(2)

TypeError: 'int' object is not iterable

### C. Write a function `mark_the_evens` that takes in a `np.array`, `arr` and returns a `np.array` of all values in `arr`, except all even numbers are replaced by `np.nan` (4 pts)

**[Double click here to see hints]**

<!---

What does `np.where` do?

--->

In [92]:
def mark_the_evens(arr):
    '''
    add docstring
    '''
    import numpy as np
    ## your code here    
    
    np.where(arr > 3)
    return arr

mark_the_evens(6)

    
    
    
    
    

6

## Q7. Approximate $\pi\$, Monte Carlo (10 points)

`numpy` is not only useful for matrix and array processing and scientific computation, it is also very useful for generating random numbers, a feature that we need for simulations. 

In this problem, we will approximate $\pi$ (which is 3.14159...) using [Monte Carlo simulations](https://en.wikipedia.org/wiki/Monte_Carlo_method) -- you may think of this method roughly as using many random samples to get an approximation. 

Now, consider a circle with diameter 1 that is inside a unit square $(0, 1)^2$. It looks like this:

<img src = "https://geometryhelp.net/wp-content/uploads/2019/08/Circle-inscribed-in-a-square.jpg" align = center, width = 200>



Let $A$ be the circle's area, and let $r$ be its radius (so $r=1/2$). Therefore the area of the circle is $A=\pi r^2$. 

However, suppose we do not know the value of $\pi$, and our task is to calculate the approximation of $\pi$. 

To do this, we can calculate $\pi$ as $A/r^2$. But, still, $A$ is unknown to us. How to address this problem? 

According to the Monte Carlo method, we can randomly sample a really large number of points ($n=10000$ at least) from within the square, then the fraction of the points landing on the circle is equal to the area of the circle. 

Following this logic, please write a program to approximate $\pi$ by first estimating the area $A$, then calculating $\pi$ as $A/r^2$. 

Note: the distance formula between two points $(x_1, y_1)$ and $(x_2, y_2)$ is $\sqrt{(x_1-x_2)^2+(y_1-y_2)^2}$.

*Again, this problem may be challenging. Please move on to other parts of the pset if you've already spent a long time on it.*

**[Double click here to see hints]**

<!---

1. Use `np.random.uniform()` to draw the coordinate (x, y) for a given random point within the unite square. `np.random.uniform()` returns a random number from 0 to 1. So you need to do the sampling twice (once for x, once for y).

2. Use np.sqrt() to calculate distance using the distance formula.

3. We know the center of the circle is (0.5, 0.5), then any point that falls within the circle must have a distance from the center less than what value?

4. Again, the area of the circle can be calculated as the proportion of the points falling inside the circle (denoted as `count`) in the total number of random points that you draw (denoted as `n`). Therefore Area = count / n. 

5. A `for` loop with a `if` statement is enough for solving the problem. 

---->


In [10]:
np.random.seed(790) 

n = 10000 # the number of random points that you would draw 

## your code below
    
    
np.random.uniform(790)    
    

    
    
    
    
    
    
    
    
    

364.1092859169652


## Q8. Big Mac Index (25 points)

Now let's move on to another important package, `pandas`. 

In this long problem, we will try to reproduce, step by step, the (unadjusted) Big Mac Index from *The Economist*, taken from this article titled [Burgernomics - The Big Mac index](https://www.economist.com/news/2020/07/15/the-big-mac-index). 

![](https://imgur.com/s8ngoz9.png)

<br>
<br>

> As shown above, the big mac index was invented by *The Economist* in 1986 as a lighthearted guide to whether currencies are at their “correct” level. It is based on the theory of purchasing-power parity (PPP), the notion that in the long run exchange rates should move towards the rate that would equalise the prices of an identical basket of goods and services (in this case, a burger) in any two countries.

The relevant data can be found at `https://raw.githubusercontent.com/TheEconomist/big-mac-data/master/source-data/big-mac-source-data.csv`. The first couple of rows of the DataFrame are shown here:

![](https://imgur.com/rRKriDN.png)

### A. Import packages, read in the data from the link above, and save it to a DataFrame sorted by date and then by country name. Display the last 5 rows of data. (3 pts)

In [24]:
## Import necessary packages here

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


## Your code here
## create new variable and dataframe
url = 'https://raw.githubusercontent.com/TheEconomist/big-mac-data/master/source-data/big-mac-source-data.csv'
df = pd.read_csv(url, header = 0)

df.sort_values(['date','name']) 
df.tail(5)

name iso_a3 currency_code  local_price   dollar_ex  GDP_dollar  \
1653        Ukraine    UKR           UAH        59.00     27.1300    3112.883   
1654        Uruguay    URY           UYU       189.00     43.6750   17014.132   
1655  United States    USA           USD         5.71      1.0000   62868.917   
1656        Vietnam    VNM           VND     66000.00  23180.0000    2551.123   
1657   South Africa    ZAF           ZAR        31.00     16.6725    6353.846   

            date  
1653  2020-07-01  
1654  2020-07-01  
1655  2020-07-01  
1656  2020-07-01  
1657  2020-07-01

### B. Create a variable `latest_date` which is equal to the most recent date that we have values for. (1 pts)

In [25]:
latest_date = df['date'].max()





In [26]:
latest_date

'2020-07-01'

### C. Convert the prices to a uniform currency. (2 pts)

Right now the `local_price` column shows the price in their own local currency, `dollar_ex` shows the exchange rate that can convert the `local_price` to the price in dollars of the Big Mac (`local_price` / `dollar_ex`). 

Using the two columns, please create a column named `dollar_price` that is  equal to `local_price` / `dollar_ex`. Then display **the last 5 rows** again. 

In [54]:
## your code here

df['dollar_ptice'] = df.groupby('local_price')['dollar_ex'].cumsum()
df.tail(5)



name iso_a3 currency_code  local_price   dollar_ex  GDP_dollar  \
1653        Ukraine    UKR           UAH        59.00     27.1300    3112.883   
1654        Uruguay    URY           UYU       189.00     43.6750   17014.132   
1655  United States    USA           USD         5.71      1.0000   62868.917   
1656        Vietnam    VNM           VND     66000.00  23180.0000    2551.123   
1657   South Africa    ZAF           ZAR        31.00     16.6725    6353.846   

            date  dollar_ptice  
1653  2020-07-01      184.1636  
1654  2020-07-01       43.6750  
1655  2020-07-01        1.0000  
1656  2020-07-01    46356.0000  
1657  2020-07-01       79.9081

### D. Create a copy of a subset of `df`, storing it in a new dataframe called `bmi` (Big Mac Index), such that it only includes countries in the `big_mac_countries` list and with `date` equal to the `latest_date` from question (B). Do this in one line. (7 pts)

**[Double click here to see hints]**

<!---

1. Recall that for a pd.Series, if you want multiple filters, you should not use `and` and `or`, but rather you should use bitwise operations. 

---->

In [109]:
## below is the list of Big mac countries.

big_mac_countries = ['ARG', 'AUS', 'BRA', 'GBR', 'CAN', 'CHL', 'CHN', 'CZE', 'DNK',
                     'EGY', 'HKG', 'HUN', 'IDN', 'ISR', 'JPN', 'MYS', 'MEX', 'NZL',
                     'NOR', 'PER', 'PHL', 'POL', 'RUS', 'SAU', 'SGP', 'ZAF', 'KOR',
                     'SWE', 'CHE', 'TWN', 'THA', 'TUR', 'ARE', 'USA', 'COL', 'CRI',
                     'PAK', 'LKA', 'UKR', 'URY', 'IND', 'VNM', 'GTM', 'HND', # Venezuela removed
                     'NIC', 'AZE', 'BHR', 'HRV', 'JOR', 'KWT', 'LBN', 'MDA', 'OMN',
                     'QAT', 'ROU', 'EUZ']



bmi = type(df.groupby(['name', 'date'])
    


SyntaxError: unexpected EOF while parsing (<ipython-input-109-b593e3b550fb>, line 14)

In [112]:
bmi

name        date
0         Argentina  2000-04-01
1         Australia  2000-04-01
2            Brazil  2000-04-01
3           Britain  2000-04-01
4            Canada  2000-04-01
...             ...         ...
1653        Ukraine  2020-07-01
1654        Uruguay  2020-07-01
1655  United States  2020-07-01
1656        Vietnam  2020-07-01
1657   South Africa  2020-07-01

[1658 rows x 2 columns]

### E. Now we will calculate the  "Big Mac Index" (6 pts) 


Please use a `for` loop to create five columns in `bmi` which show the Big Mac Index of each country in relative to the base currencies in the list `base_currencies` (['USD', 'EUR', 'GBP', 'JPY', 'CNY']) (so you will have a `USD` column, a `EUR` column, etc.) 

For each column, its row value (recall that in the `bmi` dataframe, each row denotes a country) is equal to the given country's `dollar_price` divided by the `dollar_price` of the given base currency, minus 1. 

For instance, in the `USD` column, if the row is Canada whose `dollar_price` is approximately 5.07, then the value of this cell would be 5.07 divided by 5.71 (USA's `dollar_price`) then minus 1, which is about -0.11. 

This value shows whether a country's Big Mac is undervalued or overvalued by a percentage relative to each base currency's Big Mac value. This is the so called "Big Mac Index". 

**[Double click here to see hints]**

<!---

1. For each of the base currencies, first obtain the `dollar_price` for the base currency. After that, can you create the new column in one arithmetic expression? 

---->

In [105]:
base_currencies = ['USD', 'EUR', 'GBP', 'JPY', 'CNY']

## YOUR CODE HERE
    

base_currencies = df.groupby('name')['local_price']
    
    
    
    

In [111]:
base_currencies

### F. Finally, let's visualize the Big Mac Index! (6 pts)

Use a `for` loop to generate five separate bar graphs of the Big Mac Index, one for each of the five base currencies. The X axis should show the Big Mac Index (in relative to the given base currency), the Y axis should show each country in your dataframe. 

The steps are described as below:

- First, sort your dataframe `bmi` by each base currency's Big Mac Index. 

- Then, plot the Big Mac Index using this command: `bmi.plot(kind = 'barh', x = , y = , figsize = (10, 10))` This line of code will plot the data and display the graphs for you. 

Note that: 

1. `kind = barh` plots a **horizontal** bar graph. Therefore, in your code, x should be the categorical variable (countries) and y should be the variable of interest (BMI index). 

2. For now, you don't need to worry about the titles, X/Y axis labels, and other aesthetics of the plots. A `for` loop with the above two lines of code are good enough for solving this problem. 

In [ ]:
base_currencies = ['USD', 'EUR', 'GBP', 'JPY', 'CNY']

## Your code here









## Q9. Final Project (10 points + 5 bonus points)

At the end of this pset, let's take some time to think about your final project.

### A. First, in the *Markdown* cell below, please briefly introduce your final project. (5 pts)

Specifically, write a short paragraph about:
- what question(s) you will be asking in your final project;
- what data you will be using to answer these question(s);
- where did you find your data?


For more information about the Final Project, please refer to [Blackboard > Discussions > Course Project](https://bbhosted.cuny.edu/webapps/discussionboard/do/message?action=list_messages&course_id=_1921309_1&nav=discussion_board_entry&conf_id=_2128455_1&forum_id=_2445267_1&message_id=_41966837_1)

#### This is a *Markdown* cell. Please double click this cell, and you can type in your answer here.     

Your response here:

<br>
<br>
<br>
My final project is about finding out how many minority MLB players that play in the Major Baseball prior to Jackie Robinson breaking the "color barrier" in 1947. The data used are from the variables year (specific MLB season ranging from from 1903 to 1946), ethnicity (player’s ethnicity rangning from White to Latino, to Native American) and #of_players (amount of players of a certain ethinicity). The data came from a database called data.world.












### B. Please read in your data as a `pandas` dataframe, and display the first 20 rows of your dataframe. (5 pts)

You may want to clean and format the data properly if the original dataset is large and messy (e.g. keep only the variables / subsets you need, rename the columns if necessary, etc.)

In [96]:
## import necessary packages

import pandas as pd


## your code here

pd.DataFrame(data={"Year":[1903, 1903, 1903, 1904, 1904, 1904, 1905, 1905, 1905, 1906, 1906, 1906, 1907, 1907, 1907, 1908, 1908, 1908], 
                "Ethincity":[Latino, White, Native American, Latino, White, Native American, Latino, White, Native American, Latino, White, Native American, Latino, White, Native American], 
                   "Number of Players":[0, 0.9950, 0.0050, 0, 0.9950, 0.0050, 0, 0.9950, 0.0050, 0, 0.9931, 0.0069, 0, 0.9955, 0.0045, 0, 0.9936, 0.0064]})





ValueError: arrays must all be same length

### C. Show a summary of the descriptive statistics of your data. (Optional, 5 bonus pts)

You just need to write one line to earn up to 5 points!

In [ ]:
## your code here










---
This is the end of Problem Set 1. 

*Please submit the .ipynb file via Blackboard by Oct 7, 6:30 pm.*